client/suse/compat-suse.c

```diff
@@ -1780,17 +1788,26 @@ try_infiniband(ni_suse_ifcfg_array_t *ifcfgs, ni_suse_ifcfg_t *ifcfg)
 {
 	const ni_sysconfig_t *sc = ifcfg->config;
 	ni_netdev_t *dev = ifcfg->compat->dev;
+	ni_bool_t enabled;
 	const char *umcast;
 	const char *mode;
 	const char *pkey;
+	const char *device;
 	const char *err;
 	ni_infiniband_t *ib;
 
 	mode   = ni_sysconfig_get_value(sc, "IPOIB_MODE");
 	umcast = ni_sysconfig_get_value(sc, "IPOIB_UMCAST");
+	device = ni_sysconfig_get_value(sc, "IPOIB_DEVICE");
+	pkey   = ni_sysconfig_get_value(sc, "IPOIB_PKEY");
 
-	if (!mode && !umcast && !__maybe_infiniband(dev->name))
-		return 1;
+	if (!ni_sysconfig_get(sc, "IPOIB")) {
+		if (!pkey && !device && !mode && !umcast && !maybe_infiniband(dev->name))
+			return 1;
+	} else {
+		if (!ni_sysconfig_get_boolean(sc, "IPOIB", &enabled) || !enabled)
+			return 1;
+	}
 
 	if (dev->link.type != NI_IFTYPE_UNKNOWN) {
 		ni_error("ifcfg-%s: %s config is using infiniband interface name",
```

Comment:
The "using infiniband interface name" maybe not true... this is may be better:
```suggestion
		ni_error("ifcfg-%s: %s config is contains infiniband variables",
```